src/Package/Impl/Sql/Publish/SProcScriptGenerator.cs

```diff
@@ -0,0 +1,135 @@
+        public IReadOnlyDictionary<string, string> CreateStoredProcedureScripts(SqlSProcPublishSettings settings, IEnumerable<string> sprocFiles) {
+            var dict = new Dictionary<string, string>();
+            foreach (var rFilePath in sprocFiles) {
+                var sprocName = _fs.GetSProcNameFromTemplate(rFilePath);
+                if (!string.IsNullOrEmpty(sprocName)) {
+                    string template;
+                    if (settings.CodePlacement == RCodePlacement.Inline) {
+                        template = FillSprocInlineTemplate(rFilePath, sprocName);
+                    } else {
+                        template = FillSprocTableTemplate(rFilePath, sprocName, settings.TableName, settings.QuoteType);
+                    }
+                    if (!string.IsNullOrEmpty(template)) {
+                        dict[sprocName] = template;
+                    }
+                }
+            }
+            return dict;
+        }
+
+        /// <summary>
+        /// Replaces procedure name, R Code and the SQL query placeholders with actual values
+        /// </summary>
+        private string FillSprocInlineTemplate(string rFilePath, string sprocName) {
+            var sprocTemplateFile = rFilePath.ToSProcFilePath();
+            var sprocTemplate = GetSqlFileContent(sprocTemplateFile);
+
+            var rCode = GetRFileContent(rFilePath);
+            rCode = rCode.EndsWithOrdinal(Environment.NewLine) ? rCode : rCode + Environment.NewLine;
+            sprocTemplate = sprocTemplate.Replace(RCodeTemplate, Environment.NewLine + rCode);
+
+            var sqlQuery = GetSqlFileContent(rFilePath.ToQueryFilePath()).Trim();
+            return sprocTemplate.Replace(InputQueryTemplate, sqlQuery);
+        }
+
+        private string FillSprocTableTemplate(string rFilePath, string sprocName, string codeTableName, SqlQuoteType quoteType) {
+            var sprocTemplateFile = rFilePath.ToSProcFilePath();
+            var sprocTemplate = GetSqlFileContent(sprocTemplateFile);
+
+            var format =
+@"BEGIN
+DECLARE @RCodeQuery NVARCHAR(max);
+DECLARE @RCode NVARCHAR(max);
+DECLARE @ParmDefinition NVARCHAR(max);
+
+SET @RCodeQuery = N'SELECT @RCodeOUT = RCode FROM {0} WHERE SProcName = ''{1}''';
+SET @ParmDefinition = N'@RCodeOUT NVARCHAR(max) OUTPUT';
+
+EXEC sp_executesql @RCodeQuery, @ParmDefinition, @RCodeOUT=@RCode OUTPUT;
+SELECT @RCode;
+";
+            var declarations = string.Format(CultureInfo.InvariantCulture, format, codeTableName.ToSqlName(quoteType), sprocName);
+            sprocTemplate = sprocTemplate.Replace("BEGIN", declarations);
+            sprocTemplate = sprocTemplate.Replace("N'_RCODE_'", "@RCode");
+
+            var sqlQuery = GetSqlFileContent(rFilePath.ToQueryFilePath()).Trim();
+            return sprocTemplate.Replace(InputQueryTemplate, sqlQuery);
+        }
+
+        private string GetRFileContent(string filePath) {
+            return _fs.ReadAllText(filePath).Replace("'", "''");
```

Comment:
I think no. This is to escape R code into SQL string. So in ```x <- ''``` (empty string) we do want to turn it into ```x <- ''''```